src/amber/cli/generators/api_controller.cr

```diff
@@ -13,7 +13,7 @@ module Amber::CLI
 
     private def add_routes
       add_routes :api, <<-ROUTE
-        resources "/#{name_plural}", #{class_name}Controller, except: [:new, :edit]
+        resources "/#{name_plural}", API::#{class_name}Controller, except: [:new, :edit]
```

Comment:
I'm wondering if we can specify the namespace when we perform the generate.  That way if someone doesn't want the `API` namespace or prefers a different namespace, they can specify it.

What if we support name spaces i.e. `amber generate api API::User` instead of hard coding it?

This should work for all generators so `amber g scaffold Admin::User` would create the `/admin` directory and properly create the namespaced models, controllers, views, etc.

Reply:
I can look into adding this functionality.

If (like @alex-lairan said) we are moving away from the dependence on generators, would it be better to drop the namespacing aspect in this PR altogether, and just keep the "/api" scope?

I see the merit for something like the API generator but if scaffolds are mostly going to depend on recipes, it may not be worth the effort for all other generators.

Though, the namespacing option could probably be dropped into the recipe generators too.

Reply:
I don't expect the generators will move to recipes anytime soon.  There is a lot of work needed to make them more flexible and robust.

It would be great if you could specify the namespace when generating resources.  This is a fairly big change though.  It will pretty much touch all the generated code including the tables generated for the models, etc.  The tests should help make sure you don't break existing functionality.